Fix JAMES-1815: Copy potentially immutable list to allow modifications

tfg13 opened the pull request:
Fixes https://issues.apache.org/jira/browse/JAMES-1815

chibenwa commented on server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java:
```diff
@@ -118,6 +118,9 @@ public String getDefaultDomain() throws DomainListException {
         List<String> domains = getDomainListInternal();
```
I tend to don't like variable re-affectation....

Here I would prefer a variable called "mutableDomains" to be created rather than seing domains reused.

By the way, nice spot.

tfg13 commented:
Ok, I changed the fix to introduce a new variable.
I also noticed something else: When domains is null, ImmutableList.copyof(domains) will crash with a NPE.
Also fixed that.

chibenwa commented:
:+1:

rouazana commented:
:+1:
